core/src/main/kotlin/org/eln2/space/Location.kt

```diff
@@ -134,68 +197,88 @@ enum class PlanarFace(val int: Int) {
 			})
 		}
 
+		/**
+		 * Gets the plane that has the closest normal vector to the given vector.
+		 */
 		fun fromNormal(v: Vec3i): PlanarFace = fromVec(v).inverse
 
 		/* See the microopt in Axis above */
-		val XP_VEC = Axis.X_VEC
-		val YP_VEC = Axis.Y_VEC
-		val ZP_VEC = Axis.Z_VEC
-		val XN_VEC = -XP_VEC
-		val YN_VEC = -YP_VEC
-		val ZN_VEC = -ZP_VEC
+		val PosX_VEC = Axis.X_VEC
+		val PosY_VEC = Axis.Y_VEC
+		val PosZ_VEC = Axis.Z_VEC
+		val NegX_VEC = -PosX_VEC
+		val NegY_VEC = -PosY_VEC
+		val NegZ_VEC = -PosZ_VEC
 
 		val ADJACENCIES = arrayOf(
-			arrayOf(YP, ZP, YN, ZN),
-			arrayOf(XP, ZP, XN, ZN),
-			arrayOf(XP, YP, XN, YN)
+			arrayOf(PosY, PosZ, NegY, NegZ),
+			arrayOf(PosX, PosZ, NegX, NegZ),
+			arrayOf(PosX, PosY, NegX, NegY)
 		)
 	}
 
 	val neg: Boolean get() = int > 2
 	val axis: Axis
 		get() = when (this) {
-			XP, XN -> Axis.X
-			YP, YN -> Axis.Y
-			ZP, ZN -> Axis.Z
+			PosX, NegX -> Axis.X
+			PosY, NegY -> Axis.Y
+			PosZ, NegZ -> Axis.Z
 		}
 	val inverse: PlanarFace
 		get() = when (this) {
-			XP -> XN
-			XN -> XP
-			YP -> YN
-			YN -> YP
-			ZP -> ZN
-			ZN -> ZP
+			PosX -> NegX
+			NegX -> PosX
+			PosY -> NegY
+			NegY -> PosY
+			PosZ -> NegZ
+			NegZ -> PosZ
 		}
 
-	// "Vector": the vec that points out of the block center toward this face.
+	/**
+	 "Vector": the vec that points out of the block center toward this face.
+	 */
 	val vec3i: Vec3i
 		get() = if (neg) when (axis) {
-			Axis.X -> PlanarFace.XN_VEC
-			Axis.Y -> PlanarFace.YN_VEC
-			Axis.Z -> PlanarFace.ZN_VEC
+			Axis.X -> NegX_VEC
+			Axis.Y -> NegY_VEC
+			Axis.Z -> NegZ_VEC
 		} else axis.vec3i
 
-	// "Normal": the vec that points, normal to this face, toward the block center.
+	/**
+	 * "Normal": the vec that points, normal to this face, toward the block center.
+	 */
 	val normal: Vec3i
 		get() = if (neg) axis.vec3i else when (axis) {
-			Axis.X -> PlanarFace.XN_VEC
-			Axis.Y -> PlanarFace.YN_VEC
-			Axis.Z -> PlanarFace.ZN_VEC
+			Axis.X -> NegX_VEC
+			Axis.Y -> NegY_VEC
+			Axis.Z -> NegZ_VEC
 		}
 
-	val adjacencies: Array<PlanarFace> get() = PlanarFace.ADJACENCIES[int % 3]
+	val adjacencies: Array<PlanarFace> get() = ADJACENCIES[int % 3]
 }
 
+/**
+ * A generic location in a generic space. Also contains a method to check if something can connect with this and list
+ * of locatable objects that can connect with this.
+ */
 interface Locator {
 	val vec3i: Vec3i
+
+	/**
+	 * A list of neighbors in which connections are possible with.
+	 */
 	fun neighbors(): List<Locator>
 
-	/* When overriding this, put more specific tests in more specific locators, as a rule */
+	/**
+	 * Returns true if it is possible for this node and another node to connect.
+	 * When overriding this, make sure a specific test in the locators that extend this interface.
```

Comment:
Close but not quite; the specificity is _proportional_ to the specificity _of the Locator implementation_.

If that sounds confusing, that's fine, I'll probably get around to rewording it eventually :)